### ISSUE TITLE
EXC_BAD_ACCESS on iOS build

### ISSUE THREAD

**Issue by siisee11** [homuler/MediaPipeUnityPlugin#544]
### Plugin Version or Commit ID

v0.9.1

### Unity Version

2020.3.30f1

### Your Host OS

Windows 10 Pro / Mac OS

### Target Platform

iOS

### Target Device

iPhone 7 plus

### [iOS Only] XCode Version

13.3

### Build Command

python build.py build --ios arm64 --linkopt=-s -vv

### Bug Description

App crash within 30 seconds after inferencing. It frequently occurs on iPhone 7 plus. Currently, I haven't seen this issue on iPhone 12 and 13.

Attached files are Xcode stack trace and inspector when bug occurs.

### Steps to Reproduce the Bug

1. Update mediapipe version to over 0.9.0
2. Build app and run on iPhone 8 or lower.

### Log

.

### Screenshot/Video

<img width="1300" alt="bug-2" src="https://user-images.githubusercontent.com/8529835/165552385-a7c4d14b-958d-4590-bb09-76866d97cd69.png">
<img width="676" alt="bug-1" src="https://user-images.githubusercontent.com/8529835/165552397-19bb5ddb-aa29-43a9-917f-f46355b85c6f.png">

**Comment by homuler** [comment]
Thank you for reporting the issue.

Will you test to see if this error also occurs in the official sample app (at [v0.8.9](https://github.com/google/mediapipe/releases/tag/v0.8.9))?
If it also occurs, will you further test to see if it also occurs in [the latest commit](https://github.com/google/mediapipe/commit/c6c80c37452d0938b1577bd1ad44ad096ca918e0)?

**Comment by homuler** [comment]
> 1. Update mediapipe version to over 0.9.0

I'm a little curious, does that mean it doesn't happen with older versions?

**Comment by siisee11** [comment]
@homuler  It didn't happen with 0.8.3 on iPhone 7 plus (at least it didn't crash within 10 seconds)
How can I test it? just running sample app on iPhone 7 plus?

**Comment by homuler** [comment]
> How can I test it? Just running an official sample app on iPhone 7 plus?

Yes.
Depending on the scene you tested, the app you build should be different (if you tested the Face Detection scene, then please build the Face Detection sample app).
But if it doesn't happen with v0.8.3, the built app would likely work fine.

> It didn't happen with 0.8.3 on iPhone 7 plus (at least it didn't crash within 10 seconds). I think It is a newly occurred issue. I skipped the 0.8.4 release and upgraded to 0.9.1.

I am very sorry, but if possible, I would like you to investigate which commit caused the problem using `git bisect`.
I don't have an old iPhone and have no idea what caused it, so it is difficult for me to determine the cause.

**Comment by homuler** [comment]
If it can happen with v0.8.3, then I can think of one possible cause.

In sample scenes, input `ImageFrame`s are built from `NativeArray<byte>` returned from [`Texture2D.GetRawTextureData`](https://docs.unity3d.com/ScriptReference/Texture2D.GetRawTextureData.html).
Perhaps the memory address has been changed in the middle of the process by GC, etc.

**Comment by sewonist** [comment]
I have same issue. I build and run with two devices. iPhoneXs woks find but iPhone 6s occurred above error always. 

**Plugin Version or Commit ID**
v0.9.1

**Unity Version**
2021.3.1f1

**Your Host OS**
Mac OS

**Target Platform**
iOS

**Target Device**
iPhone 6 s

**Comment by nenenkosi** [comment]
I have the same issue. I ran it on my iPhone 7 and the error occurred always when starting the app for the first time I sometimes have to restart it like 3 times for it to work.

Plugin Version or Commit ID
v0.10.0

Unity Version
2021.3.2f1

Your Host OS
Mac OS / Mac M1

Target Platform
iOS

Target Device
iPhone 7

**Comment by nenenkosi** [comment]
> I'm a little curious, does that mean it doesn't happen with older versions?

Yes I was using an older version with the same phone I didn’t get this issue

**Comment by siisee11** [comment]
Hi, homuler! I've been busy with other things lately, so I haven't been paying attention.   Is there any update related to this issue?

**Comment by homuler** [comment]
No.
As I mentioned in https://github.com/homuler/MediaPipeUnityPlugin/issues/544#issuecomment-1112074794, I don't have any devices where the issue can be reproduced.
Unless someone who has those devices identifies the cause, I don't think this issue will be solved.

**Comment by siisee11** [comment]
Hi Homuler! I built sample app of previous commits of this repo(MediaPipeUnityPlugin). This issue started to occur on  [9d7e390](https://github.com/homuler/MediaPipeUnityPlugin/commit/9d7e390b6012b2f5ebed14de599d90821f4f7f62). I hope this is useful to you 👍🏻 

If you have any idea please let me know. Thank you:)

**Comment by homuler** [comment]
@siisee11 Thank you. With https://github.com/homuler/MediaPipeUnityPlugin/commit/9d7e390b6012b2f5ebed14de599d90821f4f7f62, `GetTextureName` is not called until it is needed. If you call `TextureFrame#GetTextureName` in `TextureFrame#SetPixels32` (after [`_texture.Apply()`](https://github.com/homuler/MediaPipeUnityPlugin/commit/9d7e390b6012b2f5ebed14de599d90821f4f7f62#diff-4f083b2110991907311270d677d28380f2e0cb37658ac5a2c0934808b82c73d9R193)), does that solve the issue?

Unfortunately, I still have no way to fix it without devices.
For now, I'm thinking it is related to the behavior of `Texture2D#GetTextureNativePtr`, but we need to find out more about what is happening on the iPhone 7.
However, since this seems to be only a sample code issue, it is not a high priority at present.

**Comment by siisee11** [comment]
I tested your solution on the sample app and It didn't crash for an hour. I'll test it in my application. This is related to sample code, but I think many users of this plugin would utilize your sample code! 
I'll test further and If there is no issue I'll close this issue. 

Thank you, you are the best! 👍🏻

**Comment by homuler** [comment]
Please leave this issue open until it's fixed in the master branch.

Of course, I'd like to fix it, but in order to merge the fix, we need a rational and logical explanation of why it will solve the problem.
And as I don't have devices, I need to wait for someone to do a detailed investigation.

**Comment by siisee11** [comment]
Sure, thank you. I'll take a closer look when I have time.

**Comment by pavels-papaya** [comment]
Got the same issue, so has anything changes since then with this? Are we gonna get some fix in main branch?
